packages/bls-kms-local/src/agent/BlsKeyManagementSystem.ts

```diff
@@ -0,0 +1,97 @@
+import {blsSign, generateBls12381G2KeyPair} from '@mattrglobal/node-bbs-signatures'
+import Debug from 'debug'
+
+import {IKey, ManagedKeyInfo, MinimalImportableKey, TKeyType} from '@veramo/core'
+import {AbstractPrivateKeyStore, ManagedPrivateKey} from "@veramo/key-manager";
+import {KeyManagementSystem} from "@veramo/kms-local"
+
+const debug = Debug('veramo:kms:bls:local')
+
+export class BlsKeyManagementSystem extends KeyManagementSystem {
+
+  private readonly privateKeyStore: AbstractPrivateKeyStore;
+
+  constructor(keyStore: AbstractPrivateKeyStore) {
+    super(keyStore);
+    this.privateKeyStore = keyStore;
+  }
+
+  async importKey(args: Exclude<MinimalImportableKey, 'kms'>): Promise<ManagedKeyInfo> {
+    if (args.type === <TKeyType>'Bls12381G2') {
+      if (!args.type || !args.privateKeyHex || !args.publicKeyHex) {
+        throw new Error('invalid_argument: type, publicKeyHex and privateKeyHex are required to import a key')
+      }
+      const managedKey = this.asBlsManagedKeyInfo({ alias: args.kid, privateKeyHex: args.privateKeyHex, publicKeyHex: args.publicKeyHex, type: args.type });
+      await this.privateKeyStore.import({alias: managedKey.kid, ...args})
+      debug('imported key', managedKey.type, managedKey.publicKeyHex)
+      return managedKey
+    }
+    return super.importKey(args) as Promise<ManagedKeyInfo>
+  }
+
+  async createKey({ type }: { type: TKeyType }): Promise<ManagedKeyInfo> {
+    let key: ManagedKeyInfo
+    switch (type) {
+      case <TKeyType>'Bls12381G2': {
+        const keyPairBls12381G2 = await generateBls12381G2KeyPair()
+        key = await this.importKey({
+          kms: 'local',
+          type,
+          privateKeyHex: Buffer.from(keyPairBls12381G2.secretKey).toString("hex"),
+          publicKeyHex: Buffer.from(keyPairBls12381G2.publicKey).toString("hex"),
+        })
+        break
+      }
+      default:
+        key = await super.createKey({ type })
+    }
+
+    debug('Created key', type, key.publicKeyHex)
+
+    return key
+  }
+
+  async sign({ keyRef, algorithm, data }: { keyRef: Pick<IKey, 'kid'>; algorithm?: string; data: Uint8Array | Uint8Array[] }): Promise<string> {
```

Comment:
would be nicer to define an interface for the return and use that here instead of the whole object

Reply:
I'm using the exact interface that Veramo does. It probably makes more sense to add this changes to @veramo/key-manager and open a PR for them.

Reply:
i am talking about the return type, refactor that to an interface and use that here

Reply:
Mwah, if we want to bring this upstream, stay in line with the way upstream does it right now. Changing things like this would need to require seperate PRs